neuralprophet/df_utils.py

```diff
@@ -193,8 +193,8 @@ def data_params_definition(df, normalize, config_covariates=None, config_regress
 
     data_params = OrderedDict({})
     if df["ds"].dtype == np.int64:
-        df.loc[:, "ds"] = df.loc[:, "ds"].astype(str)
-    df.loc[:, "ds"] = pd.to_datetime(df.loc[:, "ds"])
+        df = df.assign(ds=df.loc[:, "ds"].astype(str))
+    df = df.assign(ds=pd.to_datetime(df.loc[:, "ds"]))
```

Comment:
@ourownstory what do you think about the `assign` function? According to pandas this seems to be the "official way" to assign values to a column. However, I think we slightly compromise on readability compared to `df["ds"]=...`. Should we remain with `assign()` or use the plain version?